tests/Controller/OverviewControllerTest.php

```diff
@@ -52,6 +53,88 @@ public function test_overview()
         $this->assertEquals(200, $response->getStatusCode());
 
     }
+
+    /** 
+     * Ensure that announcements can be changed by only the original user or a moderator.
+     *  
+     * @author Daniel Boling
+     */
+    public function test_modify_announcement()
+    {
+
+        $client = static::createClient();
+
+
+
+        // returns last result of all announcements
+        $announcement = static::getContainer()
+            ->get(AnnouncementRepository::class)
+            ->findAll()
+        ;
+        $announcement = end($announcement);
+        // the user should be fixture_user and the id should be 1 every time.
+
+        // return 302 since no user is authenticated
+        $client->request('GET', '/modify/announcement/' . $announcement->getId());
+        $response = $client->getResponse();
+        $this->assertEquals(302, $response->getStatusCode());
+
+        // redirect to the unauthenticated page since the wrong user is authenticated
+        $test_user = static::getContainer()
+            ->get(UserRepository::class)
+            ->findOneByUsername("test_user")
+        ;
+        $client->loginUser($test_user);
+        $client->request('GET', '/modify/announcement/' . $announcement->getId());
+        $response = $client->getResponse();
+        $this->assertEquals(403, $response->getStatusCode());
+
+        $moderator_user = static::getContainer()
+            ->get(UserRepository::class)
+            ->findOneByUsername("dboling")
+        ;
+
+        $users = array($announcement->getUser(), $moderator_user);
+        foreach ($users as $user){
+            // authenticate a fixture user and check that the page returns a 200 code
+            $client->loginUser($user);
+            $client->request('GET', '/modify/announcement/' . $announcement->getId());
+            $response = $client->getResponse();
+            $this->assertEquals(200, $response->getStatusCode());
+
+            $crawler = $client->request('GET', '/modify/announcement/' . $announcement->getId());
+            $buttonCrawlerNode = $crawler->selectButton('Submit Announcement');
+            $form = $buttonCrawlerNode->form();
+            $pre_form_fields = $form->getValues();
+
+            $announcement_subject = $pre_form_fields['announcement_form[subject]'];
+            $announcement_author = $pre_form_fields['announcement_form[author]'];
+            $this->assertSame($announcement->getSubject(), $announcement_subject);
+            $this->assertSame($announcement->getAuthor(), $announcement_author);
+
+            $client->submit($form, [
+                'announcement_form[subject]' => 'Change',
+                'announcement_form[text]' => 'change lorem ispum',
```

Comment:
It's technically possible that the values you're setting here are the same as what was already in those fields.  It's not a big deal in this instance but typically in a situation like this you would either use randomly generated text that's different each time or set it to the previous value with something appended on the end.  This is fine for what we're doing but I thought I would mention it if you run into this kind of thing in the future.